packages/cli/src/ui/components/messages/CompressionMessage.tsx

```diff
@@ -22,15 +22,29 @@ export interface CompressionDisplayProps {
 export const CompressionMessage: React.FC<CompressionDisplayProps> = ({
   compression,
 }) => {
-  const text = compression.isPending
-    ? 'Compressing chat history'
-    : `Chat history compressed from ${compression.originalTokenCount ?? 'unknown'}` +
-      ` to ${compression.newTokenCount ?? 'unknown'} tokens.`;
+  const { isPending, originalTokenCount, newTokenCount } = compression;
+
+  const originalTokens = originalTokenCount ?? 0;
+  const newTokens = newTokenCount ?? 0;
+
+  const getCompressionText = () => {
+    if (isPending) {
+      return 'Compressing chat history';
+    }
+
+    if (newTokens >= originalTokens) {
+      return 'Skipping compression for small history as the process would have increased its size.';
```

Comment:
Sadly there is a challenge with our prompt to compress chat history at the moment so sometimes very large histories may experience this issue. Due to that lets set a threshold so we only show this message if originalTokens was less than say 50,000 tokens and show a different message when originalTokens was > 5000 tokens and compression failed to actually compress.